src/source/source_failed_to_create_component_error.ts

```diff
@@ -0,0 +1,37 @@
+import { z } from "zod"
+import { getZodPrefixedIdWithDefault } from "src/common"
+import { expectTypesMatch } from "src/utils/expect-types-match"
+
+export const source_failed_to_create_component_error = z
+  .object({
+    type: z.literal("source_failed_to_create_component_error"),
+    source_failed_to_create_component_error_id: getZodPrefixedIdWithDefault(
+      "source_failed_to_create_component_error",
+    ),
+    source_component_id: z.string(),
```

Comment:
Can we add any information here? Some suggestions:
```suggestion
    constructor_name: z.string().optional(),
    properties: z.record().optional(), // NOTE: properties instead of props because we're not react specific in circuit-json land
    component_name: z.string().optional(),
    source_component_id: z.string().optional(),
```

Reply:
Why do we need `properties`?

Reply:
we probably don't- I thought it might help identify the component but component_name is probably enough. We may want to include something that can help identify the parent.

Also I'm not sure there will be a `source_component_id` if the component is never created

Reply:
e.g. we could have the standard `subcircuit_id` and potentially something like `parent_source_component_id`

Reply:
Yeah, makes sense

Reply:
@seveibar I don't think we can get the `subcircuit_id` or even the parent in fiber layer. Cause we get the subcircuit_id during the rendering process

Reply:
I think i know a hack to get it (you need to return a Renderable that is an error, ie a RenderableError), so we can add it as an optional param for now